README.md

```diff
@@ -102,6 +102,30 @@ npm install instantsearch.js algoliasearch
 yarn add instantsearch.js algoliasearch
 ```
 
+### TypeScript users
+
+To use InstantSearch in a TypeScript environment, depending on your `algoliasearch` version, different types must be imported:
+
+In case you are not using `algoliasearch` as the search client, it must still be installed in TypeScript environments.
+
+#### `algoliasearch` version 4
+
+V4 uses the types provided by both `algoliasearch` and `@algolia/client-search`.
+
+```bash
+yarn add algoliasearch@4 @algolia/client-search
```

Comment:
```suggestion
yarn add algoliasearch@4 && yarn add @algolia/client-search -D
```

Reply:
why dev dependency for that one? it's a "peer" (actually usually you won't need to rely on it manually as it's a dependency of algoliasearch), but a normal dependency would be fine imo

Reply:
I usually keep all non-runtime dependencies as dev dependencies. Not a blocker in front-end anyway, people can choose how they want they prefer.

Feel free to ignore.